UnityProject/Assets/Scripts/Objects/Research/ResearchPointMachine.cs

```diff
@@ -0,0 +1,120 @@
+using Mirror;
+using Systems.Electricity;
+using Shared.Systems.ObjectConnection;
+using UnityEngine;
+
+namespace Systems.Research.Objects
+{
+	public class ResearchPointMachine : NetworkBehaviour, IAPCPowerable, IMultitoolSlaveable, IExaminable
+	{
+		public ResearchServer researchServer;
```

Comment:
Can this only have `protected` visibility?

Reply:
I end up getting a few things complaining at me when I change it to protected. The research point methods used on `BlastYieldDetector.cs`, `ResearchServer.cs`, and `ResearchPointMachine.cs` could be rewritten in regards to this, however, `BlastYieldDetector.cs` needs to stay public for `GUI_BlastYieldDetector.cs` and I get a complaint about inconsistent accessibility from the base class of `BlastYieldDetector.cs` being less accessible.

Reply:
Unless you are refering to `public ResearchServer researchServer;`
That can be done simple enough

Reply:
I stand corrected, `GUI_BlastYieldDetector.cs` needs reference to the research server, and values from it, from `BlastYieldDetector.cs`.